bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/InstrumentDetails.scala

```diff
@@ -82,3 +82,28 @@ final case class GemsParameters(
                      avgStrehl:           Double,
                      strehlBand:          String)
 
+
+object InstrumentDetails {
+
+  // NOTE: This is similar to the code in ItcUniqueConfig which decides on imaging or spectroscopy setup
+  // on Config elements. There should be a way to share this?
```

Comment:
Unfortunately at the moment the inputs to the function are all different.  Here they are `InstrumentDetails` implementations while in Phase 2 checking they are `Config`s and elsewhere we work directly with `SPInstObsComp`s.  Even the thing being matched upon differs.  Here it is, for example for Michelle, a `String` value whereas in the case of Phase 2 checking we look for one of the `MichelleParams.Disperser` enum values.

In the new sequence model, the sequence/config and static component will be united into one immutable object.  For example see [the prototype F2 definition](https://github.com/swalker2m/ocs/blob/seqmod/bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/sequence/F2.scala). That could conceivably be used in ITC instead of the various `*Parameters` classes (?).  If so, it then it seems like it becomes possible to define these rules in one place.  Or are these initialized as servlet parameters?

Unrelated to this, I'm curious about `MichelleParameters`.   It implements, in Java, `InstrumentDetails` which is supposedly a `sealed` trait.  I'm surprised that compiles but since it surely does, I wonder if it at least issues a big warning.  If you leave `MichelleParameters` out of a match statement does it warn you about a missing case then?  I'm guessing no.